circuit-lib/src/elgamal-babyjub/elgamal.ts

```diff
@@ -0,0 +1,444 @@
+        .update(bigInt2Buffer(privKey))
+        .digest()
+        .slice(0, 32),
+    );
+    const s = ff.utils.leBuff2int(sBuff);
+    return ff.Scalar.shr(s, 3);
+  }
+
+  /**
+   * Convert a Baby Jubjub extended point into an array of two bigints.
+   */
+  export function toBigIntArray(point: BabyJubExtPoint): Array<bigint> {
+    const point_affine = point.toAffine();
+    const x = point_affine.x;
+    const y = point_affine.y;
+    return [x, y];
+  }
+
+  /**
+   * Convert an Baby Jubjub extended point into an array of two strings.
+   */
+  export function toStringArray(point: BabyJubExtPoint): Array<string> {
+    const point_affine = point.toAffine();
+    const x = point_affine.x.toString();
+    const y = point_affine.y.toString();
+    return [x, y];
+  }
+
+  /**
+   * Convert two strings x and y into a Baby Jubjub extended point.
+   */
+  export function coordinatesToExtPoint(x: string, y: string): BabyJubExtPoint {
+    const x_bigint = BigInt(x);
+    const y_bigint = BigInt(y);
+    const affine_point: BabyJubAffinePoint = { x: x_bigint, y: y_bigint };
+    const extended_point = babyJub.fromAffine(affine_point);
+    extended_point.assertValidity;
+    return extended_point;
+  }
+
+  /**
+   * Utility function used to prune a field element into a 64-bit number.
+   */
+  export function pruneTo64Bits(originalValue: bigint): bigint {
+    return originalValue & BigInt("0xFFFFFFFFFFFFFFFF");
+  }
+
+  /**
+   * Utility function used to prune a field element into a 32-bit number.
+   */
+  export function pruneTo32Bits(bigInt253Bit: bigint): bigint {
+    // Create a mask for 32 bits (all bits set to 1)
+    const mask32Bit = (1n << 32n) - 1n;
+
+    // Prune to 32 bits using the mask
+    const pruned32BitBigInt = bigInt253Bit & mask32Bit;
+
+    return pruned32BitBigInt;
+  }
+
+  /**
+   * - Returns a signal value similar to the "callGetSignalByName" function from the "circom-helper" package.
+   * - This function depends on the "circom_tester" package.
+   *
+   * Example usage:
+   *
+   * ```typescript
+   * const wasm_tester = require('circom_tester').wasm;
+   *
+   * /// the circuit is loaded only once and it is available for use across multiple test cases.
+   * const circuit = await wasm_tester(path.resolve("./circuit/path"));
+   * const witness = await circuit.calculateWitness(inputsObject);
+   * await circuit.checkConstraints(witness);
+   * await circuit.loadSymbols();
+   *
+   * /// You can check signal names by printing "circuit.symbols".
+   * /// You will mostly need circuit inputs and outputs.
+   * const singalName = 'main.out'
+   * const signalValue = getSignalByName(circuit, witness, SignalName)
+   * ```
+   */
+  export const getSignalByName = (
+    circuit: any,
+    witness: any,
+    signalName: string,
+  ) => {
+    const signal = `main.${signalName}`;
+    return witness[circuit.symbols[signal].varIdx].toString();
+  };
+}
+
+export namespace DecodeUtils {
```

Comment:
what do you think about moving this namespace into a separate file?

Reply:
moved namespace into src/elgamal-babyjub/decode.ts